### ISSUE TITLE
Announce TOG classes

### ISSUE THREAD

**Issue by edreamleo** [leo-editor/leo-editor#1565]
This will complete #1440.

- [x] Support stand-alone operation.
- [x] Write an announcement.  See the second comment.
- [x] Announce on the main python announcement list.
- [x] Announce at [Python issue 33337](https://bugs.python.org/issue33337).
- [x] Email to ast tokens people.

**Status**

The leoAst project is a complete success.The code in leoAst.py is simpler, easier to understand, more flexible, more robust and faster than the corresponding code in the asttokens, fstringify and black projects.

**Comparison with asttokens**

Most python users will find [asttokens](https://github.com/gristlabs/asttokens) meets all their needs. asttokens has:
- excellent [documentation](https://asttokens.readthedocs.io/en/latest/user-guide.html).
- an excellent [api](https://asttokens.readthedocs.io/en/latest/api-index.html).
- elegant and compact code.

I studied the sources for asttokens both before and after working on this project. The asttokens code solves all the problems I had to solve, but the asttokens code is much more compact, and more general.

Here are the opening comments (really a docstring) from MarkTokens._visit_after_children, in asttokens.mark_tokens.py:
```python
# This processes the node generically first, after all children have been processed.

# Get the first and last tokens that belong to children. Note how this doesn't assume that we
# iterate through children in order that corresponds to occurrence in source code. This
# assumption can fail (e.g. with return annotations).
```
In other words, the asttokens code does *not* depend on token-order traversal.

**Comment by edreamleo** [comment]
**Draft Announcement**

The Theory of Operation describes the TOG and TOT classes.

token order traversal is a fundamental resource.

I admire asttokens tool. It will often be the best tool for the job.

The asttokens code and TOG classes have different notions of elegance.

black is splits and joins lines well. Alas, it messes with comments.

Leo's fstringify commands used to be superior to fstringify tool. This may have changed.

Leo's beautify and fstringify commands formed a proof of concept for the TOG classes.

**Comment by edreamleo** [comment]
This third comment contains a draft of the announcement to python devs.

**Comment by edreamleo** [comment]
This is the [fourth comment]() of Leo issue #1565.

**Comment by edreamleo** [comment]
I am reopening this issue because it contains a draft proposal to the python devs.
